sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -287,7 +286,7 @@ class Analyzer(
           case (_, CalendarIntervalType) => Cast(TimeSub(l, r), l.dataType)
           case (TimestampType, _) => SubtractTimestamps(l, r)
           case (_, TimestampType) => SubtractTimestamps(l, r)
-          case (_, DateType) => if (conf.usePostgreSQLDialect) {
+          case (_, DateType) => if (conf.ansiEnabled) {
```

Comment:
Actually `SubtractDates` is the ansi behavior.

Since `date - date` is newly added in 3.0, we can just change this to `case (_, DateType) => SubtractDates(l, r)`

Reply:
Copy that, then we need to change the postgreSQL/date.sql.out correspondingly. Done in 17a3c77.